test/test_game.py

```diff
@@ -0,0 +1,221 @@
+import unittest
+
+from rock_paper_scissors.core.exc import InvalidRoundsError, InvalidChoiceError
+from rock_paper_scissors.core.game import Game
+from rock_paper_scissors.core.moves import Paper, Rock
+from rock_paper_scissors.core.user import User
+from rock_paper_scissors.core.valid_choices import VALID_CHOICES
+
+
+class GameInitTestCase(unittest.TestCase):
+    def test_player1_and_player2_and_starting_rounds_types(self):
+        player1 = 'Joe'
+        player2 = 'Jane'
+        game = Game(player1, player2)
+
+        self.assertIsInstance(game.player1, str)
+        self.assertIsInstance(game.player2, str)
+        self.assertIsInstance(game._rounds, int)
+        self.assertEqual(game._rounds, 0)
+
+
+class GameSetRoundsTestCase(unittest.TestCase):
+    def setUp(self):
+        self.game = Game('player1', 'player2')
+
+    def test_valid_rounds_input(self):
+        value = self.game.rounds = 11
+
+        self.assertIsInstance(value, int)
+        self.assertGreater(value, 0)
+
+    def test_raises_error_if_round_not_an_int(self):
+        with self.assertRaises(InvalidRoundsError):
+            self.game.rounds = 'a'
+
+    def test_raises_error_if_value_is_not_odd(self):
+        with self.assertRaises(InvalidRoundsError):
+            self.game.rounds = 2
+
+    def test_raises_error_if_value_is_not_positive(self):
+        with self.assertRaises(InvalidRoundsError):
+            self.game.rounds = -1
+
+
+class GameGetRoundsTestCase(unittest.TestCase):
+    def test_return_rounds_of_game(self):
+        self.game = Game('player1', 'player2')
+        self.game._rounds = 1
+
+        self.assertEqual(self.game.rounds, 1)
+
+
+class GameValidChoicesTestCase(unittest.TestCase):
+    def test_assert_tuple_of_valid_choices(self):
+        self.game = Game('player1', 'player2')
+        game_valid_choices = self.game.valid_choices
+
+        self.assertIsInstance(game_valid_choices, tuple)
+        self.assertCountEqual(game_valid_choices, tuple(VALID_CHOICES.keys()))
+
+
+class GameConvertPlayerInputToMove(unittest.TestCase):
+    def test_user_input_is_a_valid_string_for_conversion_to_move(self):
+        self.game = Game('player1', 'player2')
+        expected = VALID_CHOICES['rock']
+        result = self.game.convert_input_to_move('rock')
+
+        self.assertEqual(expected, result)
+
+    def test_user_input_is_not_a_valid_string_for_conversion_to_object(self):
+        self.assertRaises(InvalidChoiceError, Game.convert_input_to_move, 'fish')
+
+
+class GamePlayLogicTestCase(unittest.TestCase):
+    def setUp(self):
+        player1 = User('Joe')
+        player2 = User('Jane')
+        self.game = Game(player1, player2)
+
+    def test_player1_move_equals_player2_move_returns_tie(self):
+        self.game.player1.choice = Rock()
+        self.game.player2.choice = Rock()
+
+        actual_tie = self.game.play()
+        expected_tie = 'Tie'
+
+        self.assertIsInstance(actual_tie, str)
```

Comment:
Technically you don't need this assertion, since `assertEqual` will fail if `actual_tie` is not a string.